### ISSUE TITLE
GitHub webhooks seem to have no effect

### ISSUE THREAD

**Issue by oliversalzburg** [Strider-CD/strider#720]
I recently updated our Strider instance, since then pushing to GitHub no longer causes Strider to update.

I even removed all of our projects and re-created them. Still, I can only manually test and deploy.

**Comment by knownasilya** [comment]
@oliversalzburg try updating all plugins, then making sure that your github auth works. You might have to remove your github account, and re-add it to strider.

**Comment by oliversalzburg** [comment]
@knownasilya All plugins are updated. I had removed and re-added my account when I removed all projects.

**Comment by knownasilya** [comment]
Do you see any errors? Do you have a firewall that's been brought up?

**Comment by oliversalzburg** [comment]
I don't think it's anything related to networking.

When I push to GitHub, I see nothing in the strider log. The web UI also doesn't show me anything.

I just realized there are some red triangles in the /settings/hooks page on GitHub. I'll check those real quick.

**Comment by oliversalzburg** [comment]
Ugh, okay, this is definitely my fault :P Thanks anyway

![](http://i.imgur.com/OCjtCMs.png)

**Comment by oliversalzburg** [comment]
http://serverfault.com/questions/665606/why-does-nginx-redirect-to-urls-with-a-trailing-slash if anyone has any ideas ;D

**Comment by oliversalzburg** [comment]
@knownasilya After further investigation, it seems like this might not be an nginx issue after all. 

When I simply run a GET with curl on the webhook URL, I also get a 301 logged in my strider log.

**Comment by knownasilya** [comment]
This might be fixed with the recent PRs, could you try master?

**Comment by knownasilya** [comment]
If that doesn't work, I have another idea, https://github.com/Strider-CD/strider/blob/314ef56bdd051e142f697643e9ea2e95e85af14c/lib/utils/load-extensions.js#L64

express 4 probably doesn't need this.

**Comment by oliversalzburg** [comment]
I pulled the latest master, but the issue was unchanged.

I also removed the `slashes()` line you mentioned, now I don't get a 301 reply when I GET the URL (I receive a 404 instead), but the POST from GitHub still fails with a 301.

I'll take a closer look at this next week and untangle nginx from our setup once more. At this point I'm not really sure yet where the problem comes from, nginx or strider. Stupidly, I updated strider and set up the nginx reverse proxy at the same time.
